runtime/src/comm/ofi/comm-ofi-oob-pmi2.c

```diff
@@ -286,6 +291,51 @@ void decode_kvs(char* raw, const char* enc, size_t size) {
 }
 
 int chpl_comm_ofi_oob_locales_on_node(void) {
-  // TODO: figure out how to implement this correctly
-  return 1;
+  int count = 0;
+  if (PMI_Get_clique_size != NULL) {
+    PMI_CHK(PMI_Get_clique_size(&count));
+    DBG_PRINTF(DBG_OOB, "PMI_Get_clique_size returned %d", count);  
+  } else {
+    // do an allgather of hostname hashes to determine the locales on the same node as us
+    // assumes each hostname has a unique hash
+
+    char hostname[256];
+    int rc = gethostname(hostname, sizeof(hostname));
```

Comment:
`chpl_nodeName` uses `uname` instead of `gethostname`. I wonder if there's a portability reason to prefer this? Looks like the `uname` was originally added in https://github.com/chapel-lang/chapel/commit/b58a4e7884 and adapted in https://github.com/chapel-lang/chapel/commit/ffa3842222. @bradcray @gbtitus do you see any reason to prefer one or the other?

Reply:
Actually, this file already includes chplsys.h, so @jhh67 I think we should just use `chpl_nodeName`

Reply:
Huh, I didn't get a notification for your original @-tag for some reason... I don't know of any reason to prefer either of `gethostname` vs. `uname`.  Given the timeframe of these commits, I wouldn't be surprised if I was simply more familiar with `uname`.

Reply:
chpl_nodeName doesn't work because it calls chpl_mem_realloc and the memory allocator isn't initialized yet. I don't think there is a difference between gethostname and uname, I believe on linux they both use the gethostname system call. I've always used the former.

Reply:
I've unresolved this conversation just to add that at least on our Linux systems, the `uname(3p)` man page doesn't give much guidance as to which might be preferable, but the `uname(2)` man page recommends `gethostname()` as being more meaningful. Instead of a hardcoded size of 256 it would be better to use `HOST_NAME_MAX+1`, though. (`+1` because the `gethostname(3p)` man page says that "host names are limited to [that size]", but doesn't say whether that includes the trailing `'\0'` byte or not.)